RT_UPDATE/download_nLx.R

```diff
@@ -0,0 +1,416 @@
+          ", years ",
+          paste(nLxDatesIn, collapse = ", "),
+          ", gender ",
+          gender
+        ),
+        sep = "\n"
+      )
+    }
+    
+    # standardize input strings to match what we expect
+    sex_code <- ifelse(tolower(gender) == "both",
+                       "b",
+                       ifelse(
+                         tolower(gender) == "female",
+                         "f",
+                         ifelse(tolower(gender) == "male", "m", NA)
+                       ))
+    
+    # sex standardization
+    Sex_mortlaws <- ifelse(sex_code == "b", "total", tolower(gender))
+    
+    stopifnot(`Invalid sex name, please set it to 'both', 'male' or 'female'` = !is.na(sex_code))
+    
+    # I take the single year mx data, filter country, sex and years up to the latest wpp 
+    # then I sue interp for each combination of country and sex for nLxDatesIn (all years used)
+    # and thne I calculate the lt_single_mx for same combination
+    out <- mx1dt %>%
+      as_tibble() %>%
+      filter(.data$country_code %in% location_code,
+             .data$year < parse_number(latest_wpp) + 1) %>%
+      pivot_longer(-c("country_code", "name", "year", "age"),
+                   names_to  = "sex",
+                   values_to = "mx") %>%
+      mutate(sex = str_remove(.data$sex, "mx"), 
+             sex = tolower(.data$sex)) %>%
+      filter(.data$sex %in% sex_code) %>%
+      pivot_wider(names_from  = year, 
+                  values_from = mx) %>%
+      select(-"age") %>%
+      group_nest(.data$country_code, .data$name, .data$sex) %>%
+      # interpolate
+      mutate(data = map(
+        data,
+        ~ interp(
+          .x,
+          as.numeric(names(.x)),
+          as.numeric(nLxDatesIn),
+          extrap = TRUE,
+          method = method,
+          ...
+        ) %>%
+          as_tibble()
+      )) %>% 
+      unnest("data") %>% 
+      pivot_longer(-c("country_code", "name", "sex"),
+                   names_to  = "year",
+                   values_to = "mx") %>% 
+      group_nest(.data$country_code, .data$name, .data$sex, .data$year) %>%
+      # calculate lifetable
+      # SRB???
+      mutate(data = map2(.x = data, 
+                         .y = sex, ~ lt_single_mx(nMx    = .x$mx,
+                                                  radix  = radix,
+                                                  Sex    = .y,
+                                                  a0rule = "ak")))
+    
+    # now if the output is single ages, we do not need no extra
+    if(output == "single") {
+      
+      out <- out %>% 
+        unnest("data") %>%
+        select(age = "Age", "year", "nLx") %>% 
+        # wide format
+        pivot_wider(names_from  = year, 
+                    values_from = nLx) %>% 
+        column_to_rownames("age") %>% 
+        as.matrix()
+      
+    }
+    
+    # if output is abridged, then we calculate abridged LT from single with DemoTools
+    if(output == "abridged") {
```

Comment:
ok, this works either way; just need two ways to have previously derived out 1. from single age mx, 2. from abridged mx or 5-y mx.